src/azure-cli/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -153,6 +153,28 @@
 phone_extension_type = CLIArgumentType(
     help='The extension of the customer service number of the publisher. Only digits are allowed and number of digits should not exceed 10.')
 
+kind_type = CLIArgumentType(
+    help="The kind of topic resource.",
+    arg_type=get_enum_type(['azure', 'azurearc']),
+    options_list=['--kind'],
+    is_preview=True
+)
+
+extended_location_name = CLIArgumentType(
+    help="The extended location name if kind==azurearc.",
+    options_list=['--extended-location-name'],
+    arg_group="Azure Arc",
+    is_preview=True
+)
+
+extended_location_type = CLIArgumentType(
+    help="The extended location type if kind==azurearc.",
+    arg_type=get_enum_type(['customlocation']),
```

Comment:
Same here.

Another question is if `extended_location_type` only have one allowed value, can this parameter be omitted? We can always set it `customlocation` in `custom.py`.

Reply:
yes i thought about that. but customers don't get that option while invoking REST API so thought just doing something in CLI will break that flow.